src/components/MySpace/MySpace.tsx

```diff
@@ -100,109 +115,141 @@ const MySpace = ({ bookmarks }: { bookmarks: BookmarkRecords }) => {
 
           {data &&
             data.mySpace &&
-            data.mySpace.map((widget: Widget) => (
-              <Grid
-                key={`widget_${widget._id}`}
-                tabletLg={{ col: 6 }}
-                desktopLg={{ col: 4 }}>
-                {widget.type === 'News' && <NewsWidget widget={widget} />}
-
-                {isCollection(widget) && (
-                  <CustomCollection
-                    _id={widget._id}
-                    title={widget.title}
-                    type={widget.type}
-                    bookmarks={widget.bookmarks || []}
-                    bookmarkOptions={bookmarks}
-                    handleRemoveCollection={() => {
-                      handleRemoveCollection({
-                        variables: {
-                          _id: widget._id,
-                        },
-                        refetchQueries: [`getMySpace`],
-                      })
-                    }}
-                    handleEditCollection={(
-                      title: string,
-                      bookmarks?: Bookmark[]
-                    ) => {
-                      handleEditCollection({
-                        variables: {
-                          _id: widget._id,
-                          title,
-                          bookmarks,
-                        },
-                        optimisticResponse: {
-                          editCollection: {
+            data.mySpace.map((widget: Widget) => {
+              if (
+                widget.type === 'GuardianIdeal' &&
+                flags.guardianIdealCarousel
+              ) {
+                return (
+                  <Grid
+                    key={`widget_${widget._id}`}
+                    className={styles.guardianIdeal}>
+                    <GuardianIdealCarousel
+                      ideals={GuardianIdealPillars}
+                      widget={widget}
+                    />
+                  </Grid>
+                )
+              }
+
+              if (widget.type === 'News') {
+                return (
+                  <Grid
+                    key={`widget_${widget._id}`}
+                    tabletLg={{ col: 6 }}
+                    desktopLg={{ col: 4 }}>
+                    <NewsWidget widget={widget} />
+                  </Grid>
+                )
+              }
+
+              if (isCollection(widget)) {
+                return (
+                  <Grid
+                    key={`widget_${widget._id}`}
+                    tabletLg={{ col: 6 }}
+                    desktopLg={{ col: 4 }}>
+                    <CustomCollection
```

Comment:
@gidjin @abbyoung  remember earlier when we were discussing how sometimes we just do things because they make sense? This is one of those moments. When I moved the LaunchDarkly flag, I had to write a questionable ternary to check if the widget is of type `GuardianIdeal` or not. It was really sloppy, and I found this whole section to be confusing, so I decided to rewrite it. I think that this new pattern will make it easier to add new widgets to My Space, as well as place any of them behind a feature flag. Bonus points: it helps with styling too, because you can wrap any widget (new or old) in any flavor of `<Grid />` you want, and it all looks clean! Also, thinking ahead to drag-and-drop of widgets, I think this new approach will make that a far less confusing experience. Let me know what you think!

Reply:
This is great! I agree this is much more readable than the ternary that was there before. Thank you for doing this. Also seems clear enough now if we decided to refactor it to a higher order component or something it will be simpler.